nav.go

```diff
@@ -195,86 +198,61 @@ func newDir(path string) *dir {
 	}
 }
 
-func normalize(s1, s2 string, ignorecase, ignoredia bool) (string, string) {
-	if ignorecase {
-		s1 = strings.ToLower(s1)
-		s2 = strings.ToLower(s2)
-	}
-	if ignoredia {
-		s1 = removeDiacritics(s1)
-		s2 = removeDiacritics(s2)
-	}
-	return s1, s2
-}
-
 func (dir *dir) sort() {
 	dir.sortby = getSortBy(dir.path)
 	dir.dirfirst = getDirFirst(dir.path)
 	dir.dironly = getDirOnly(dir.path)
 	dir.hidden = getHidden(dir.path)
 	dir.reverse = getReverse(dir.path)
+
 	dir.hiddenfiles = gOpts.hiddenfiles
 	dir.ignorecase = gOpts.ignorecase
 	dir.ignoredia = gOpts.ignoredia
 
+	collopts := []collate.Option{}
+	if dir.ignorecase {
+		collopts = append(collopts, collate.IgnoreCase)
+	}
+	if dir.ignoredia {
+		collopts = append(collopts, collate.IgnoreDiacritics)
+	}
+	if dir.sortby == naturalSort {
+		collopts = append(collopts, collate.Numeric)
+	}
+	coll := collate.New(language.English, collopts...)
+
 	dir.files = dir.allFiles
 
-	// reverse order cannot be applied after stable sorting, otherwise the order
-	// of equivalent elements will be reversed
+	var lessfun func(i, j int) bool
+
 	switch dir.sortby {
 	case naturalSort:
-		sort.SliceStable(dir.files, func(i, j int) bool {
-			s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
-			if !dir.reverse {
-				return naturalLess(s1, s2)
-			} else {
-				return naturalLess(s2, s1)
-			}
-		})
+		lessfun = func(i, j int) bool {
+			return coll.CompareString(dir.files[i].Name(), dir.files[j].Name()) == -1
+		}
 	case nameSort:
-		sort.SliceStable(dir.files, func(i, j int) bool {
-			s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
-			if !dir.reverse {
-				return s1 < s2
-			} else {
-				return s2 < s1
-			}
-		})
+		lessfun = func(i, j int) bool {
+			return coll.CompareString(dir.files[i].Name(), dir.files[j].Name()) == -1
+		}
```

Comment:
These two cases have the same code, so you should be able to combine them:

```go
	case nameSort, naturalSort:
		lessfun = func(i, j int) bool {
			return coll.CompareString(dir.files[i].Name(), dir.files[j].Name()) == -1
		}
```

It's probably also worth putting a comment stating that `naturalSort` is already handled above.

This actually raises the question of whether natural sorting should have its own option, instead of being lumped under `sortby`, since it is really a sorting modifier and not a property (e.g. name, size, time) that is used as a basis for sorting. But I think it is not worth worrying about for now.